esphome/components/i2c/__init__.py

```diff
@@ -62,6 +63,7 @@ def _bus_declare_type(value):
             ),
             cv.Optional(CONF_TIMEOUT): cv.positive_time_period,
             cv.Optional(CONF_SCAN, default=True): cv.boolean,
+            cv.Optional(CONF_REPORT_ACTIVITY, default=True): cv.boolean,
```

Comment:
Even the config and code-gen could be done directly from Component schema, but un-intuitive which component implementations that actually has this reporting implemented.

Reply:
Also, `default=True` is likely not the wanted behavior